ledger/accountdb.go

```diff
@@ -860,7 +2071,8 @@ func (qs *accountsDbQueries) lookup(addr basics.Address) (data persistedAccountD
 			data.addr = addr
 			if len(buf) > 0 && rowid.Valid {
 				data.rowid = rowid.Int64
-				return protocol.Decode(buf, &data.accountData)
+				err = protocol.Decode(buf, &data.accountData)
```

Comment:
It seems like either this change is unneeded (because directly returning the error will cause it to be propagated to the outer `err` variable by `db.Retry`, or if this change IS needed (because that doesn't happen), then a similar change is needed on line 2084.

My guess is that returning the error directly is ok. But I'm pretty sure they should be done the same way.

Reply:
this change allow you to place a breakpoint on the `return err` so you could observe the generated error. looking on your comment, I was worried about shadowed variables, but that was never the case here.